FuncGameDev/FSharp/Unity/Update.fs

```diff
@@ -36,14 +42,43 @@ type Updater() =
     member this.checkGameOver() =
         match GameState.instance.gamedata.floor with 
         | 11 ->
-            match highScore with
-            | highScore when highScore |> float > GameState.instance.gamedata.time ->
-                ScoreSavingService.storeScore (GameState.instance.gamedata.time |> string)
-            | _ -> ()
+            match highestRemaining with
```

Comment:
It seems in this case we're saving an incorrect highest floor, since we never update it. Pulling this all out into another function would help this as well.